lib/rules/no-use-before-define.js

```diff
@@ -105,6 +153,10 @@ module.exports = function(context) {
      */
     function findVariablesInScope(scope) {
         scope.references.forEach(function(reference) {
+            if (reference.init) {
```

Comment:
I'm thinking...

```js
var a = a;
```

In this case, escope makes 2 references. 

- One is the write reference of the first `a` for initialization.
- One is the read reference of the second `a`.

This rule does not need to handle the former, so it exits here.

Hmm, the comment might be "Does not need to handle references of initialization.".